scripts/omni_run.py

```diff
@@ -2,220 +2,184 @@
 
-    uv.data_array = np.expand_dims(np.array([data[p] for p in pols]).T.swapaxes(0, 1), axis=1)
-    uv.vis_units = 'uncalib'
-    uv.nsample_array = np.ones_like(uv.data_array, dtype=np.float)
-    uv.flag_array = np.zeros_like(uv.data_array, dtype=np.bool)
-    uv.Nspws = 1  # this is always 1 for paper and hera(currently)
-    uv.spw_array = np.array([uv.Nspws])
-    blts = np.array([bl for bl in bls for i in range(uv.Ntimes)])
-    if xtalk:
-        uv.time_array = np.array(list(m['times'][:1]) * uv.Nbls)
-        uv.lst_array = np.array(list(m['lsts'][:1]) * uv.Nbls)
-    else:
-        uv.time_array = np.array(list(m['times']) * uv.Nbls)
-        uv.lst_array = np.array(list(m['lsts']) * uv.Nbls)
-
-    # generate uvw
-    uvw = []
-    for t in range(uv.Ntimes):
-        for bl in bls:
-            uvw.append(aa.gen_uvw(*bl, src='z').reshape(3, -1))
-    uv.uvw_array = np.array(uvw).reshape(-1, 3)
-
-    uv.ant_1_array = blts[:, 0]
-    uv.ant_2_array = blts[:, 1]
-    uv.baseline_array = uv.antnums_to_baseline(uv.ant_1_array,
-                                               uv.ant_2_array)
-
-    uv.freq_array = m['freqs'].reshape(1, -1) * 1e9  # Turn into MHz.
-    poldict = {'xx': -5, 'yy': -6, 'xy': -7, 'yx': -8}
-    uv.polarization_array = np.array([poldict[p] for p in pols])
-    if xtalk:
-        # xtalk integration time is averaged over the whole file
-        uv.integration_time = m2['inttime'] * len(m2['times'])
-    else:
-        uv.integration_time = m2['inttime']
-    uv.channel_width = np.diff(uv.freq_array[0])[0]
-
-    # observation parameters
-    uv.object_name = 'zenith'
-    uv.telescope_name = 'HERA'
-    uv.instrument = 'HERA'
-    tobj = pyuvdata.uvtel.get_telescope(uv.telescope_name)
-    uv.telescope_location = tobj.telescope_location
-    uv.history = m2['history']
-
-    # phasing information
-    uv.phase_type = 'drift'
-    uv.zenith_ra = uv.lst_array
-    uv.zenith_dec = np.array([aa.lat] * uv.Nblts)
-
-    # antenna information
-    uv.Nants_telescope = 128
-    uv.antenna_numbers = np.arange(uv.Nants_telescope, dtype=int)
-    uv.Nants_data = len(np.unique(np.concatenate([uv.ant_1_array, uv.ant_2_array]).flatten()))
-    uv.antenna_names = np.array(['ant{0}'.format(ant) for ant in uv.antenna_numbers])
-    antpos = []
-    for k in aa:
-        antpos.append(k.pos)
+args = np.sort(args)
+pols = opts.pol.split(',')
 
-    uv.antenna_positions = np.array(antpos)
+if len(args)==0: raise AssertionError('Please provide visibility files.')
+if opts.minV and len(list(set(''.join(pols))))==1: raise AssertionError('Stokes V minimization requires crosspols in the "-p" option.')
 
-    if returnuv:
-        return uv
-    else:
-        print('   Saving {0}'.format(filename))
-        uv.write_uvfits(filename, force_phase=True, spoof_nonessential=True)
+def getPol(fname): return fname.split('.')[3] #XXX assumes file naming format
+def linPol(polstr): return len(list(set(polstr)))==1
```

Comment:
Is `linPol` the right name for this function? It makes me think "linear polarization", which I don't think is what it's testing. If I understand the function, it's testing whether a polarization string is XX or YY rather than, e.g. XY. MIRIAD calls these "intensity-type" polarizations which I kind of like.